Refactor main CLI method

NopAngel opened the pull request:
Hello, *spenserblack*!

This is my pull request fixing the `src/cli.rs` file (basically, an optimization)

Main improvements:

- [x] I deleted `should_edit_file()`: The function was unnecessary since it only checked if edit_config was Some. Now I use pattern matching directly in run().

- [x] I extracted `run_tree()`: I separated the main tree logic into its own function to improve readability.

- [x] I reused `lua_state.to_inner()`: Instead of calling it three times, I store it in a local variable.

- [x] I simplified tree construction: I used `map` and `unwrap_or` to apply configurations conditionally in a more idiomatic way.

- [x] I passed `edit_config` as a parameter: Instead of extracting it twice (in `should_edit_file` and `edit_file`), I pass it directly.

- [x] I renamed variables for clarity: `contents` → `default_contents` to better indicate their purpose.

- [x] I reduced nesting: The logic for `color_choice` is now flatter.
- [x] I removed unnecessary comments: The `TODO` statement about "skip loading config" is no longer relevant with the current structure.


I hope this is helpful, have a great day/afternoon/evening!

spenserblack commented on src/cli.rs:
```diff
@@ -46,14 +46,19 @@ impl Cli {
         let git = Git::new(&self.path).expect("Should be able to read the git repository");
 
-        // NOTE The Lua state must live as long as the configuration values.
+        // Build Lua state with git if available
```
```suggestion
        // NOTE The Lua state must live as long as the configuration values.
```
Why delete this note? It helps explain that `mlua` can act in surprising ways for Rust users. For example, if a `mlua::Function` lives longer than the `mlua::Lua` that created it, calling that function could panic.

spenserblack commented on src/cli.rs:
```diff
@@ -62,77 +67,67 @@ impl Cli {
 
-        let color_choice = self
-            .color_choice
+        // Determine color choice with less nesting
```
```suggestion
```

spenserblack commented on src/cli.rs:
```diff
@@ -62,77 +67,67 @@ impl Cli {
+        
         let tree = builder.build();
 
+        // Execute in git scope
```
```suggestion
```

Self-documenting IMO

spenserblack commented on src/cli.rs:
```diff
@@ -62,77 +67,67 @@ impl Cli {
         };
 
-        // NOTE If we can't check if it exists, we'll be safe and skip overwriting it.
+        // Create file if it doesn't exist
```
```suggestion
        // NOTE If we can't check if it exists, we'll be safe and skip overwriting it.
```
This note explains why we're just ignoring the try exists error.

spenserblack commented on src/cli.rs:
```diff
@@ -62,77 +67,67 @@ impl Cli {
 
         let finder = find_editor::Finder::with_extra_environment_variables([Self::EDITOR_ENV_VAR]);
-        /// Should the program wait for the editor to close before continuing?
         const WAIT: bool = true;
```
Admittedly, this doc comment is pretty unnecessary, but I'd hesitate to remove comments in a "refactor" PR.

```suggestion
        /// Should the program wait for the editor to close before continuing?
        const WAIT: bool = true;
```

spenserblack reviewed (changes requested):
Looks good! Most of my requests are just for adding back comments that were removed.

No offense, but was this done with AI? If you're generating code with AI, I'd encourage you to double-check to confirm that it's matching the style of the codebase (e.g. it dropped the comment prefixes that I like to use).

NopAngel commented:
Hello, good morning/evening! xd (Merry Christmas, btw). The code (the one I sent) didn't use AI, buttttt I later sent it to an AI to have my corrections made (i.e., it added comments and stuff xd) because I don't usually use AI in my code. 🤓👆

NopAngel commented:
@spenserblack 

I've fixed the problems you mentioned:

- [x] I fixed the compilation error: I simply replaced the problematic lines with `if let` to avoid ownership issues.

- [x] I ran `cargo fmt`: it's now formatted.

- [x] I kept all the comments with `NOTE:` and `TODO:` prefixes as you requested.

The main changes I kept from my original refactor:

- I separated `run_tree()` into its own function

- I reused `lua_state.to_inner()` in a local variable

- I passed `edit_config` directly as a parameter.

I compiled on my machine and it worked quite well; in theory, it should compile fine on all machines. Thanks for the detailed review!
